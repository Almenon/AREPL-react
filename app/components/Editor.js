import React, {Component} from 'React'

export default class Editor extends Component<Props>{
    props: Props;

    render(){
        return <div style={{width:'400px',height:'800px',backgroundColor:'#d6d1d1'}}></div>
    }
}