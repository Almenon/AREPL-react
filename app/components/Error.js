import React, {Component} from 'React'

export default class Error extends Component<Props>{
    props: Props;

    render(){
        return <div style={{width:'200px',height:'200px',color:'red'}}>oh noes an error!</div>
    }
}