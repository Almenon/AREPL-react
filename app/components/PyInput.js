// @flow
import React, {Component} from 'react'

type Props = {
    onInput:Function
}
type State = {
    value:string
}

export default class PyInput extends Component<Props,State>{
    props: Props
    state = {value: '>> input'}
  
    handleChange = ({target}:SyntheticInputEvent<>) =>{
        //handleChange is not invoked upon enter so no logic here
        this.setState({value: target.value});
    }

    handleKeyPress = (event:SyntheticKeyboardEvent<>)=>{
        if(event.key == "Enter"){
            this.props.onInput(this.state.value)
            this.setState({value: ""});
        }
    }
  
    render() {
      return <input style={{display:"inherit"}} id="stdin" type="text" value={this.state.value} onChange={this.handleChange} onKeyPress={this.handleKeyPress}/>;
    }
  }

/* NOTES
tried typing event with SyntheticKeyboardEvent<HTMLInputElement>
but flow still didn't like event.target.value :(
see https://github.com/facebook/flow/issues/368

when first developing this i made mistake of assigning state directly - but that doesnt trigger re-render of component
i also ran into binding issues ~ see https://medium.freecodecamp.org/react-binding-patterns-5-approaches-for-handling-this-92c651b5af56
*/