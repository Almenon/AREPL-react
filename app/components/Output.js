import React, {Component} from 'React'
import Error from './Error'
import VariableView from './VariableView'
import Stdout from './Stdout'

type Props = {
    output:{
        vars: {},
        errMessage: string,
        execTime: string,
        totalTime: string
    }
}

export default class Output extends Component<Props>{
    props: Props;

    render(){
        return <div>
            <Error errorMessage={this.props.output.errMessage}></Error>
            <VariableView vars={this.props.output.vars}></VariableView>
            <Stdout></Stdout>
        </div>
    }
}