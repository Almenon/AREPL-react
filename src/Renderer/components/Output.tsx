import * as React from 'React'
import Error from './Error'
import Stdout from './Stdout'
import { connect } from 'react-redux';
import { RootState } from '@red/index';
import ReactJson from 'react-json-view'
import { areplState } from '@red/arepl';

type Props = {
    output:areplState
}

export class Output extends React.Component<Props>{

    render(){
        return <div>
            <Error errorMessage={this.props.output.userError}/>
            <ReactJson src={this.props.output.userVariables} collapseStringsAfterLength={90} displayDataTypes={false} name={false} enableClipboard={false} collapsed={2} style={{fontSize:"16px"}}/>
            <Stdout lines={this.props.output.stdout}/>
        </div>
    }
}

function mapStateToProps(state: RootState) {
    return {
        output: state.arepl
    }
  }
  
export default connect(mapStateToProps, ()=>{})(Output as any);