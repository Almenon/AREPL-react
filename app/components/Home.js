// @flow
import React, { Component } from 'react';
//import { Link } from 'react-router-dom';
import PyInput from './PyInput'
import Output from './Output'
import styles from './Home.css';
import Editor from './Editor';

type Props = {
  onCode: Function
};

export default class Home extends Component<Props> {
  props: Props;

  render() {
    return (
      <div style={{display:'flex',flexDirection:'row'}}>
        <div className={styles.editorGroup}>
          <Editor></Editor>
          <PyInput onCode={this.props.onCode}></PyInput>
        </div>
        <div className={styles.output}>
          <Output></Output>
        </div>
      </div>
    );
  }
}

// <Link to="/counter">to Counter</Link>
// that's if i wanted to do a link
// ill probably use a link for a settings view