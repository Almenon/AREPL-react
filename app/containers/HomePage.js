// @flow
import React, { Component } from 'react';
import Home from '../components/Home';
import {executeCode} from '../actions/arepl'
import {connect} from 'react-redux'

const mapStateToProps = (state, ownProps) => {
  return {}
}

const mapDispatchToProps = (dispatch, ownProps) => {
  return {
    onCode: code => dispatch(executeCode(code)),
    onInput: input => console.log(input)
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Home)
