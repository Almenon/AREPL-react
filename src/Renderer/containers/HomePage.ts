import Home from '../components/Home';
import {executeCode} from '../actions/arepl'
import {connect} from 'react-redux'

const mapStateToProps = (state:any, ownProps:any) => {
  return {}
}

const mapDispatchToProps = (dispatch:Function, ownProps:any) => {
  return {
    onCode: (code:{savedCode:string,evalCode:string}) => dispatch(executeCode(code)),
    onInput: (input:string) => console.log(input)
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Home)
