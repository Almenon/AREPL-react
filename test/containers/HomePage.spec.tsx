import * as React from 'react'
import Enzyme, { mount } from 'enzyme'
import  { Provider } from 'react-redux'
import {createBrowserHistory} from 'history'
import { ConnectedRouter } from 'react-router-redux'
import HomePage from '../../src/Renderer/containers/HomePage'
import configureStore from '../../src/Renderer/store/configureStore.dev'
import { Output } from '../../src/Renderer/components/Output';

interface MockEvent{
    target:{
        value:string
    }
}

function setup(initialState?){
    const store = configureStore.configureStore(initialState)
    const history = createBrowserHistory();
    const provider = (
        <Provider store={store}>
            <ConnectedRouter history={history}>
                <HomePage/>
            </ConnectedRouter>
        </Provider>
    )
    const app = mount(provider)
    return {
        app,
        editor: app.find('textarea'),
        error: app.find('Error'),
        output: app.find('Output'),
        varDisplay: app.find('VariableView')
    }
}

describe('HomePage', ()=>{
    it('should output vars when user types in editor', () => {
        const {editor, output, varDisplay} = setup()

        const mockEvent: MockEvent = {
            target: {
                value: "x=1"
            }
        }

        editor.prop('onCode')
        editor.simulate('change', mockEvent)

        expect(output.props()).toMatchSnapshot()
        expect(varDisplay.prop('vars')).toMatchSnapshot()
    })
})