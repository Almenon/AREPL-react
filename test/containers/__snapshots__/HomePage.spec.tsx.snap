// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage should output vars when user types in editor 1`] = `
<VariableView
  vars={Object {}}
>
  <div
    style={
      Object {
        "height": "200px",
        "width": "200px",
      }
    }
  >
    {}
  </div>
</VariableView>
`;

exports[`HomePage should output vars when user types in editor 2`] = `Object {}`;
